src/WebJobs.Extensions.EasyTables/Bindings/EasyTableAttributeBindingProvider.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.WindowsAzure.MobileServices;
+using WebJobs.Mobile.EasyTables;
+
+namespace WebJobs.Extensions.EasyTables
+{
+    internal class EasyTableAttributeBindingProvider : IBindingProvider
+    {
+        private EasyTableConfiguration _easyTableConfig;
+        private JobHostConfiguration _jobHostConfig;
+        private INameResolver _nameResolver;
+
+        public EasyTableAttributeBindingProvider(JobHostConfiguration config, EasyTableConfiguration easyTableConfig, INameResolver nameResolver)
+        {
+            _jobHostConfig = config;
+            _easyTableConfig = easyTableConfig;
+            _nameResolver = nameResolver;
+        }
+
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            ParameterInfo parameter = context.Parameter;
+            EasyTableAttribute attribute = parameter.GetCustomAttribute<EasyTableAttribute>(inherit: false);
+            if (attribute == null)
+            {
+                return Task.FromResult<IBinding>(null);
+            }
+
+            if (_easyTableConfig.MobileAppUri == null)
+            {
+                throw new InvalidOperationException(
+                    string.Format(CultureInfo.CurrentCulture,
+                    "The Easy Tables Uri must be set either via a '{0}' app setting or directly in code via EasyTableConfiguration.MobileAppUri.",
+                    EasyTableConfiguration.AzureWebJobsMobileAppUriName));
+            }
+
+            EasyTableContext easyTableContext = CreateContext(_easyTableConfig, attribute, _nameResolver);
+
+            IBindingProvider compositeProvider = new CompositeBindingProvider(new IBindingProvider[]
```

Comment:
That work may help in the future. GenericBinder only works for output bindings and greatly simplified my output binding support. But they don't address input bindings right now.

I can write my own CompositeBindingProvider if that's the concern. It's simple and in fact I already did this until I realized this already existed (Paul suggested I use the existing one in a past CR).

What I want is a way to let the Bindings tell me what they support and retrieve the first one that matches the param type. That keeps all the if/else branching out of this method -- I did that at first and it was very ugly and tough to follow.


---
In reply to: [54810717](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/45#discussion_r54810717) [](ancestors = 54810717)